### ISSUE TITLE
the value format option should accept Python f-strings

### ISSUE THREAD

**Issue by aqc-carlodri** [holoviz/panel#5928]
#### Is your feature request related to a problem? Please describe.
Following the documentation for the [Gauge indicator](https://panel.holoviz.org/reference/indicators/Gauge.html), I was trying to use the `format` option to reduce the precision of the display of a floating point value. Reading the docs, I expected that this was a standard Python f-string formatting, but it turns out it is not, and there is no simple way I could find to obtain the desired display.

[x-ref Discord thread](https://discord.com/channels/1075331058024861767/1179053116193247252)

#### Describe the solution you'd like
I would like (and expect) this format to be a Python f-string for maximum flexibility on the representation of the `value`. Just as an example, I would like to be able to write something like `format = "{value:0.2f} MPa"`

#### Describe alternatives you've considered
Not that I could think of.

**Comment by jbednar** [comment]
Agreed. In this case I'd also suggest that the default precision of the spinner is much too high. I don't think any decimal points of precision make sense for most applications; a number 0 to 100 is precision enough for nearly any application!

**Comment by aqc-carlodri** [comment]
One thought concerning the title of the issue: maybe this is not confined to the `Gauge` indicator but to any other place where value formatting can be relevant. But I haven't explored Panel enough to tell at the moment.

**Comment by jbednar** [comment]
My understanding is that the gauge implementation is separate and needs special handling, but if it can be done at a more general level, sure!
